Automattic/wp-calypso
# Welcome Tour: Refactor and extract core functionality to /packages

## What this is about

This PR is Part 1 (alpha) of extracting the tour component into a publishable package for NPM.

Extracts the core functionality of the welcome tour and wraps it into a standalone, configurable package under `/packages` for general reuse. Bare-bones, nothing really styled. This would be equivalent to what we'd get with a Shepherd or other tour package, only more React-aware.

- Offers a simple API via configuration for enabling/disabling various features
- By default, the accessibility functionalities are provided: minimize/close on Esc, left-right key navigation

Check the [REAME.md](https://github.com/Automattic/wp-calypso/tree/add/welcome-tour-extract/packages/packaged-tour) for the more detailed overview. 

## Next up / on the horizon

We plan on extracting the rest of the welcome tour into a component for reuse also. That would include all the current setup - styling, controls, skeleton, minimized view, etc. To be decided whether the related APIs & data stores are also to be extracted and how.

## Testing

1. Ensure that the current welcome tour works as expected. Nothing should change functionality or design-wise.
    - `yarn install; yarn dev --sync` from `/apps/editing-tookit`
    - go to the editor, sidebar, and open the welcome guide
    - adding `&welcome-tour-next` to the URL should show the guided-tour with spotlight effects
2. Check the storybook examples demo ok: `yarn run packaged-tour:storybook:start`

## Next

Some follow-up issues for improving various parts will follow in the respective board.

### for shipping:

- [x] Improve README
- [x] Add storybook
- [x] A name (`packaged-tour` is a placeholder) 😎 
- [x] More testing, debugging, cleanup.

<img width="943" alt="Screenshot 2021-11-24 at 9 15 23 AM" src="https://user-images.githubusercontent.com/1705499/143192450-a7213708-30af-4323-bcbf-3d8e2dbe8d69.png">

## Media

**Storybook:**

![Kapture 2021-11-23 at 16 25 22](https://user-images.githubusercontent.com/1705499/143042285-9608fe3a-33d5-4dc2-943b-257b34d85be4.gif)

no flag | `&welcome-tour-next`
-- | --
<img width="606" alt="Screenshot 2021-11-23 at 4 26 30 PM" src="https://user-images.githubusercontent.com/1705499/143043045-56cf5ac5-59f1-46fc-85d6-745ed6d7d78d.png"> | <img width="716" alt="Screenshot 2021-11-23 at 4 25 58 PM" src="https://user-images.githubusercontent.com/1705499/143043111-80fa75b1-840d-4a22-847b-73e135d86437.png">
